### DISCUSSION TITLE
Movement duration statistics

### DISCUSSION THREAD

**Discussion by nsuprin** [Sibyx/phpGPX#73] [Ideas]
Hello Everyone,
It should be a great improvement to be able to get the informations about the movement duration.
In many GPX files, we can observe some time gap wich are often a real pause for the persons wich are recording the track on the terrain.
And today, the duration is simply calculated by a diff between last point timestamp and first point timestamp.
So the avgSpeed and Pacce are impacted negativly.
It should be more accurate to distinguish the global duration, and the movement only duration.
THx

**Comment by Sibyx** [comment]
Sup @nsuprin!

This is now implemented in the upcoming 2.x release on the `develop` branch. The new single-pass Engine includes a `MovementAnalyzer` that computes exactly what you described:

- `Stats::$movingDuration` — time spent actually moving (excluding pauses)
- `Stats::$movingAverageSpeed` — average speed based on moving time only

It works by comparing instantaneous speed between consecutive points against a configurable speed threshold (default 0.5 m/s). You can tune it via the engine factory:

```php
$gpx = new phpGPX(engine: Engine::default(
    speedThreshold: 1.0, // m/s — adjust to your needs
));
$file = $gpx->load('track.gpx');

$stats = $file->tracks[0]->stats;
echo $stats->duration;           // total duration
echo $stats->movingDuration;     // movement-only duration
echo $stats->movingAverageSpeed; // speed without pauses
```

If you're still interested, feel free to check out the `develop` branch.
